app.py

```diff
@@ -174,47 +143,59 @@ async def update_cl(config: dict, settings: dict):
 # https://docs.chainlit.io/examples/qa
 @cl.on_chat_start
 async def on_chat_start():
-    "Send a chat start message to the chat and load the model config."
+    """
+    Triggered at the start of a chat session. It loads the model configuration from a file
+    and sets it in the user session for future use.
+    """
     config = load_config()
     cl.user_session.set("config", config)
-    await update_cl(config, None)
+    await update_cl(config, {})
 
-    await cl.Message(content=config["introduction"], disable_feedback=True).send()
+    await cl.Message(content=config["introduction"]).send()
 
 
 @cl.on_message
 async def on_message(message: cl.Message):
-    """Handle a message.
+    "Chat message handler."
+    runnable = cl.user_session.get("chain")
+    msg = cl.Message(content="")
+
+    class PostMessageHandler(BaseCallbackHandler):
```

Comment:
Rather than handling the message using the chainlit backend, literal AI have opted to require langchain users to write a message handler directly into their on_message. This is ripped from their docs and modified to work with my source metadata. It's not ideal as it doesn't link to the direct source text, so I will file an issue for this.